source/Posets/Delta-V-CompletePoset.lagda

```diff
@@ -0,0 +1,206 @@
+
+\begin{code}
+
+δ_𝓥-complete-Poset : (𝓤 𝓦 𝓥 : Universe) → (𝓤 ⁺) ⊔ (𝓦 ⁺) ⊔ (𝓥 ⁺) ̇
+δ_𝓥-complete-Poset 𝓤 𝓦 𝓥 = Σ A ꞉ Poset 𝓤 𝓦 , δ_𝓥-complete-poset.is-δ_𝓥-complete 𝓥 A
+
+Poset-δ : (D : δ_𝓥-complete-Poset 𝓤 𝓦 𝓥) → Poset 𝓤 𝓦
+Poset-δ D = pr₁ D
+
+is-δ_𝓥-complete-δ : (D : δ_𝓥-complete-Poset 𝓤 𝓦 𝓥) →
+  δ_𝓥-complete-poset.is-δ_𝓥-complete 𝓥 (Poset-δ D)
+is-δ_𝓥-complete-δ D = pr₂ D
+
+\end{code}
+
+\begin{code}
+
+module non-trivial-posets {𝓤  𝓦  𝓥 : Universe} (A : Poset 𝓤 𝓦) where
+ is-non-trivial-poset : 𝓤 ⊔ 𝓦 ̇
+ is-non-trivial-poset =  Σ x ꞉ ∣ A ∣ₚ ,
+  ( Σ y ꞉ ∣ A ∣ₚ , (x ≤[ A ] y) holds × ¬ (x ＝ y))
+
+ lower : is-non-trivial-poset → ∣ A ∣ₚ
+ lower i = pr₁ i
+
+ upper : is-non-trivial-poset → ∣ A ∣ₚ
+ upper i = pr₁ (pr₂ i)
+
+ order : (i : is-non-trivial-poset) → (lower i ≤[ A ] upper i) holds
+ order i = pr₁ (pr₂ (pr₂ i))
+
+ nequal : (i : is-non-trivial-poset) → ¬ (lower i ＝ upper i)
+ nequal i = pr₂ (pr₂ (pr₂ i))
+
+ open Joins (rel-syntax A)
+ open δ_𝓥-complete-poset 𝓥 A
+
+ wlem-lemma : (i : is-non-trivial-poset) → (P : Ω 𝓥) →
+  (((lower i) is-lub-of ((𝟙 + P holds) , (δ (lower i) (upper i) P))) holds → ¬ (P holds)) ×
+  (((upper i) is-lub-of ((𝟙 + P holds) , (δ (lower i) (upper i) P))) holds → ¬ ¬ (P holds)) 
+ pr₁ (wlem-lemma i P) r p = nequal i (≤-is-antisymmetric A (order i) (pr₁ r (inr p)))
+ pr₂ (wlem-lemma i P) r np = nequal i (≤-is-antisymmetric A (order i) (pr₂ r (( lower i , h ))))
+  where
+   h : ((lower i) is-an-upper-bound-of ((𝟙 + P holds) , δ (lower i) (upper i) P)) holds
+   h (inl ✯) = ≤-is-reflexive A (lower i)
+   h (inr p) = 𝟘-induction (np p)
+    
+\end{code}
+
+We now show that the two element poset is δ_𝓥 complete only if WLEM holds.
+
+\begin{code}
+
+2-is-non-trivial : non-trivial-posets.is-non-trivial-poset 2-Poset
+2-is-non-trivial = (₀ , ₁ , ⋆ , zero-is-not-one)
+
+2-is-δ_𝓥-complete-WLEM : {𝓥 : Universe} →
+ δ_𝓥-complete-poset.is-δ_𝓥-complete {𝓤₀} {𝓤₀} 𝓥 2-Poset →
+ (P : Ω 𝓥) → is-decidable (¬ (P holds))
+2-is-δ_𝓥-complete-WLEM {𝓥} i P = decide-¬P
+ where
+  open Joins (rel-syntax 2-Poset)
+  open δ_𝓥-complete-poset 𝓥 2-Poset
+  open non-trivial-posets 2-Poset
+
+  h : Σ s ꞉ ∣ 2-Poset ∣ₚ , (s is-lub-of ((𝟙 + P holds) , (δ ₀ ₁ P))) holds
+  h = i ₀  ₁ ⋆ P
+
+  g : Σ s ꞉ ∣ 2-Poset ∣ₚ , (s is-lub-of ((𝟙 + P holds) , (δ ₀ ₁ P))) holds →
+    is-decidable (¬ (P holds))
+  g (₀ , sup) = inl (pr₁ (wlem-lemma 2-is-non-trivial P) sup)
+  g (₁ , sup) = inr (pr₂ (wlem-lemma 2-is-non-trivial P) sup)
+
+  decide-¬P : is-decidable (¬ (P holds))
+  decide-¬P = g h
+
+\end{code}
+
+Since non-trivial is a negated concept it only has enough strength to derive WLEM, we now introduce the stronger concept of positivity.
+
+\begin{code}
+
+module positive-posets {𝓤  𝓦  𝓥 : Universe} (A : Poset 𝓤 𝓦) where
+ open Joins (rel-syntax A)
+ open δ_𝓥-complete-poset 𝓥 A
+ open Universal fe
+ open PosetReasoning A
+
+ _is-strictly-below_ : (x y : ∣ A ∣ₚ) → {i : is-δ_𝓥-complete} → 𝓤 ⊔ 𝓦 ⊔ (𝓥 ⁺) ̇ 
```

Comment:
With my version of Agda (2.6.3), I'm getting quite a few unresolved implicit arguments, because of `is-strictly-below`'s `{i : is-δ_𝓥-complete}`. I suggest you make this a module parameter before defining `is-strictly-below` to resolve this. This also relates to your comment at the end.

Reply:
I noticed the unresolved arguments! Thanks for the fix!